rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -178,72 +179,75 @@ lv2_fs_mount_point* lv2_fs_object::get_mp(std::string_view filename)
+	if (mp_name.starts_with("dev_usb"sv))
+		return &g_mp_sys_dev_usb;
+	if (mp_name == "dev_bdvd"sv)
+		return &g_mp_sys_dev_bdvd;
+	if (mp_name == "dev_ps2disc"sv)
+		return &g_mp_sys_dev_dvd;
+	if (mp_name == "app_home"sv)
+		return &g_mp_sys_app_home;
+	if (mp_name == "host_root"sv)
+		return &g_mp_sys_host_root;
+	if (mp_name == "dev_flash"sv)
+		return &g_mp_sys_dev_flash;
+	if (mp_name == "dev_flash2"sv)
+		return &g_mp_sys_dev_flash2;
+	if (mp_name == "dev_flash3"sv)
+		return &g_mp_sys_dev_flash3;
 
 	// Default fallback
-	return &g_mp_sys_dev_root;
+	return &g_mp_sys_no_device;
 }
 
 std::string lv2_fs_object::get_vfs(std::string_view filename)
 {
 	const auto mp_name = get_device_path(filename);
 
-	if (!mp_name.empty())
-	{
-		if (mp_name == "dev_hdd0"sv)
-			return g_cfg_vfs.get(g_cfg_vfs.dev_hdd0, rpcs3::utils::get_emu_dir());
-		if (mp_name == "dev_hdd1"sv)
-			return g_cfg_vfs.get(g_cfg_vfs.dev_hdd1, rpcs3::utils::get_emu_dir());
-		if (mp_name.starts_with("dev_usb"sv))
-			return g_cfg_vfs.get_device(g_cfg_vfs.dev_usb, fmt::format("/%s", mp_name), rpcs3::utils::get_emu_dir()).path;
-		if (mp_name == "dev_bdvd"sv)
-			return g_cfg_vfs.get(g_cfg_vfs.dev_bdvd, rpcs3::utils::get_emu_dir());
-		if (mp_name == "dev_ps2disc"sv)
-			return {}; // Unsupported in VFS
-		if (mp_name == "app_home"sv && filename.data() != Emu.argv[0].data())
-			return lv2_fs_object::get_vfs(Emu.argv[0]);
-		if (mp_name == "host_root"sv)
-			return g_cfg.vfs.host_root ? "/" : std::string();
-		if (mp_name == "dev_flash"sv)
-			return g_cfg_vfs.get_dev_flash();
-		if (mp_name == "dev_flash2"sv)
-			return g_cfg_vfs.get_dev_flash2();
-		if (mp_name == "dev_flash3"sv)
-			return g_cfg_vfs.get_dev_flash3();
-
-		// Default
+	if (mp_name == "dev_hdd0"sv)
 		return g_cfg_vfs.get(g_cfg_vfs.dev_hdd0, rpcs3::utils::get_emu_dir());
-	}
+	if (mp_name == "dev_hdd1"sv)
+		return g_cfg_vfs.get(g_cfg_vfs.dev_hdd1, rpcs3::utils::get_emu_dir());
+	if (mp_name.starts_with("dev_usb"sv))
+		return g_cfg_vfs.get_device(g_cfg_vfs.dev_usb, fmt::format("/%s", mp_name), rpcs3::utils::get_emu_dir()).path;
+	if (mp_name == "dev_bdvd"sv)
+		return g_cfg_vfs.get(g_cfg_vfs.dev_bdvd, rpcs3::utils::get_emu_dir());
+	if (mp_name == "dev_ps2disc"sv)
+		return {}; // Unsupported in VFS
+	if (mp_name == "app_home"sv)
+		return g_cfg_vfs.get(g_cfg_vfs.app_home, rpcs3::utils::get_emu_dir());
+	if (mp_name == "host_root"sv)
+		return g_cfg.vfs.host_root ? "/" : std::string();
+	if (mp_name == "dev_flash"sv)
+		return g_cfg_vfs.get_dev_flash();
+	if (mp_name == "dev_flash2"sv)
+		return g_cfg_vfs.get_dev_flash2();
+	if (mp_name == "dev_flash3"sv)
+		return g_cfg_vfs.get_dev_flash3();
 
 	// Default fallback
 	return {};
 }
 
+s32 lv2_fs_object::get_mount_count()
+{
+	s32 count = std::popcount(g_mp_sys_dev_usb.is_mounted);
+	lv2_fs_mount_point* mp = &g_mp_sys_dev_root;
+
+	while (mp)
+	{
+		if (mp != &g_mp_sys_dev_usb && mp->is_mounted != 0U)
+			count++;
+		mp = mp->next;
```

Comment:
This is a for loop body, it just that instead of increment you assign a pointer.

Reply:
That's neat!